ablateLibrary/eos/stiffenedGas.cpp

```diff
@@ -0,0 +1,102 @@
+    }
+    ke *= 0.5;
+
+    // assumed eos
+    (*internalEnergy) = (totalEnergy)-ke; // not correct maybe?
+    *p = (parameters->gamma - 1.0) * density * (*internalEnergy) - parameters->gamma * parameters->p0;
+    *a = PetscSqrtReal(parameters->gamma * ((*p) + parameters->p0) / density);
+    PetscFunctionReturn(0);
+}
+
+PetscErrorCode ablate::eos::StiffenedGas::StiffenedGasComputeTemperature(PetscInt dim, PetscReal density, PetscReal totalEnergy, const PetscReal *massFlux, const PetscReal densityYi[], PetscReal *T,
+                                                                     void *ctx) {
+    PetscFunctionBeginUser;
+    Parameters *parameters = (Parameters *)ctx;
+
+    // Get the velocity in this direction
+    PetscReal speedSquare = 0.0;
+    for (PetscInt d = 0; d < dim; d++) {
+        speedSquare += PetscSqr(massFlux[d] / density);
+    }
+
+    // assumed eos
+    PetscReal internalEnergy = (totalEnergy)-0.5 * speedSquare; // is this right?
+    PetscReal cv = parameters->Cv; // rGas / (parameters->gamma - 1.0);
+
+    (*T) = (internalEnergy - parameters->e0) / cv + parameters->T0;
+    PetscFunctionReturn(0);
+}
+
+PetscErrorCode ablate::eos::StiffenedGas::StiffenedGasComputeSpeciesSensibleEnthalpy(PetscReal temperature, PetscReal *hi, void *ctx) {
+    PetscFunctionBeginUser;  // do I need to make new one, or reuse perfectGas function ????
+    Parameters *parameters = (Parameters *)ctx;
+    for (PetscInt s = 0; s < parameters->numberSpecies; s++) {
+        hi[s] = 0.0;
+    }
+    PetscFunctionReturn(0);
+}
+PetscErrorCode ablate::eos::StiffenedGas::StiffenedGasComputeDensityFunctionFromTemperaturePressure(PetscReal temperature, PetscReal pressure, const PetscReal *yi, PetscReal *density, void *ctx) {
+    PetscFunctionBeginUser;
+    Parameters *parameters = (Parameters *)ctx;
+    *density = (pressure + parameters->gamma * parameters->p0) / ((parameters->gamma - 1) * (parameters->Cv * (temperature - parameters->T0) + parameters->e0) );
+    PetscFunctionReturn(0);
+}
+PetscErrorCode ablate::eos::StiffenedGas::StiffenedGasComputeSensibleInternalEnergy(PetscReal T, PetscReal density, const PetscReal *yi, PetscReal *sensibleInternalEnergy, void *ctx) {
+    PetscFunctionBeginUser;
+    Parameters *parameters = (Parameters *)ctx;
+    PetscReal cv = parameters->Cv; // rGas / (parameters->gamma - 1.0);
+    *sensibleInternalEnergy = (T - parameters->T0) * cv + parameters->e0;
+    PetscFunctionReturn(0);
+}
+PetscErrorCode ablate::eos::StiffenedGas::StiffenedGasComputeSpecificHeatConstantPressure(PetscReal T, PetscReal density, const PetscReal *yi, PetscReal *specificHeat, void *ctx) {
+    PetscFunctionBeginUser; // don't know how to calculate this at the moment
+    Parameters *parameters = (Parameters *)ctx;
+    (*specificHeat) = parameters->gamma * parameters->Cv; // parameters->gamma * parameters->rGas / (parameters->gamma - 1.0);
+            // maybe 6150.0  ??? (does not add up to water props for ref state)
```

Comment:
Wikipedia says that it mispredicts this for water, so I don't think we would expect it compute it correctly.